version: '3'

services:

  redis:
    image: redis
    ports:
      - '6380:6379'


  db:
    image: postgres
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=Antoniya2308
      - PGDATA=/var/lib/postgresql/data/pgdata
      - POSTGRES_DB=coursework_7

    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
        - '5432:5432'
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5


  app:
    build: .
    tty: true
    command: bash -c "python3 manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/Habits_tracker_Docker
    ports:
      - '8000:8000'
    depends_on:
      db:
        condition: service_healthy


  celery:
      build: .
      tty: true
      env_file:
        - .env
      command: celery -A config worker -l INFO
      depends_on:
        - redis
        - app


  celery_beat:
    build: .
    tty: true
    env_file:
      - .env
    restart: always
    command: celery -A config beat -l INFO -S django
    depends_on:
      - redis
      - app


volumes:
  pg_data: